From b3b9808808ef7315d5623b60054534107cd22895 Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Mon, 9 Mar 2020 18:35:35 +0700
Subject: [PATCH] arm64: dts: renesas: hihope-common: Enable Audio support in
 Low speed connector

This patch enables sound support in Low speed connector
for the common HiHope G2M and G2N boards by adding
device node of the PCM5102 codec (External I2S Board
in Low speed Connector).

Therefore, we support 2 audio codecs in common plarform:
1. HDMI audio from built-in HDMI transmitter. (default)
2. External I2S Audio Codec PCM5102A in Low Speed Connector.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 arch/arm64/boot/dts/renesas/hihope-common.dtsi | 37 ++++++++++++++++++++++++--
 1 file changed, 35 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/hihope-common.dtsi b/arch/arm64/boot/dts/renesas/hihope-common.dtsi
index 5efec7b..45a6891 100644
--- a/arch/arm64/boot/dts/renesas/hihope-common.dtsi
+++ b/arch/arm64/boot/dts/renesas/hihope-common.dtsi
@@ -92,7 +92,8 @@
 
 		label = "rcar-sound";
 
-		dais = <&rsnd_port0>;
+		dais = <&rsnd_port0	/* HDMI Audio */
+			&rsnd_port1>;	/* External PCM5102A Codec */
 	};
 
 	vbus0_usb2: regulator-vbus0-usb2 {
@@ -175,6 +176,17 @@
 		#clock-cells = <0>;
 		clock-frequency = <24000000>;
 	};
+
+	pcm5102a: pcm5102a {
+		compatible = "ti,pcm5102a";
+		#sound-dai-cells = <0>;
+
+		port {
+			pcm5102a_endpoint: endpoint {
+				remote-endpoint = <&rsnd_endpoint1>;
+			};
+		};
+	};
 };
 
 &audio_clk_a {
@@ -343,6 +355,11 @@
 		power-source = <1800>;
 	};
 
+	sound_pins: sound {
+		groups = "ssi01239_ctrl", "ssi0_data", "ssi1_data_a";
+		function = "ssi";
+	};
+
 	sound_clk_pins: sound_clk {
 		groups = "audio_clk_a_a";
 		function = "audio_clk";
@@ -392,7 +409,7 @@
 };
 
 &rcar_sound {
-	pinctrl-0 = <&sound_clk_pins>;
+	pinctrl-0 = <&sound_pins &sound_clk_pins>;
 	pinctrl-names = "default";
 
 	status = "okay";
@@ -412,9 +429,25 @@
 				playback = <&ssi2>;
 			};
 		};
+
+		rsnd_port1: port@1 {
+			rsnd_endpoint1: endpoint {
+				remote-endpoint = <&pcm5102a_endpoint>;
+
+				dai-format = "i2s";
+				bitclock-master = <&rsnd_endpoint1>;
+				frame-master = <&rsnd_endpoint1>;
+
+				playback = <&ssi1 &src1 &dvc0>;
+			};
+		};
 	};
 };
 
+&ssi1 {
+	shared-pin;
+};
+
 &rwdt {
 	timeout-sec = <60>;
 	status = "okay";
-- 
2.7.4

